components/script/dom/gamepadbuttonlist.rs

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::codegen::Bindings::GamepadButtonListBinding;
+use dom::bindings::codegen::Bindings::GamepadButtonListBinding::GamepadButtonListMethods;
+use dom::bindings::js::{JS, Root};
+use dom::bindings::reflector::{Reflector, reflect_dom_object};
+use dom::gamepadbutton::GamepadButton;
+use dom::globalscope::GlobalScope;
+use dom_struct::dom_struct;
+use webvr_traits::WebVRGamepadButton;
+
+// https://www.w3.org/TR/GamepadButton/
+#[dom_struct]
+pub struct GamepadButtonList {
+    reflector_: Reflector,
+    list: Vec<JS<GamepadButton>>
+}
+
+impl GamepadButtonList {
+    #[allow(unrooted_must_root)]
+    fn new_inherited(list: Vec<JS<GamepadButton>>) -> GamepadButtonList {
+        GamepadButtonList {
+            reflector_: Reflector::new(),
+            list: list
+        }
+    }
+
+    #[allow(unrooted_must_root)]
+    pub fn new(global: &GlobalScope, list: Vec<Root<GamepadButton>>) -> Root<GamepadButtonList> {
```

Comment:
Is this method even needed? I can't see it called anywhere. If it's not, just remove it.